### PR TITLE
Refactor: colony home page

### LINKED ISSUE #1517: Refactor colony home route
Currently the home route does not have its own context provider, so a lot of things are mixed left and right.
We should create a context provider for it, so we can handle opening the leave colony modal there, simplifying the dashboard header component, use composition instead of using this config approach.
image

### PR DESCRIPTION
**Pull request by bassgeta**
## Description

Refactored the ColonyHome route so it now uses its own context provider :)
I had to name the context `ColonyDashboardContext` since `ColonyHomeContext` exists, but I didn't need any of those values.
If you have any better naming ideas, feel free to ping me and let's change em!
I've tried untangling the `useDashboardHeader` hook as much ass possible, but the `DropdownMenu` component is using objects and not components, so I wasn't able to completely rework everything, so that bit of code smell is now encapsulated in the `useHeaderLinks` hook.

I also took the liberty to move the entire component into `v5/frame` as it's a page not a `common` component.

**New stuff** ✨

* ColonyDashboard context provider

**Changes** 🏗

* Took out  leave colony modal to a separate component'
* Moved the entire folder into the `v5/frame` directory
* External links are now rendered via JSX directly (not in the dropdown though) 

Resolves #1517

### CODE REVIEW

**Comment by jakubcolony** [comment]
@bassgeta looks like it needs a rebase

**Review by ArmandoGraterol — DISMISSED** [review]
Looks reasonable to me 👍 Nice job untangling all of that

File: src/components/v5/frame/ColonyHome/partials/DashboardHeader/partials/ColonyLinks/useHeaderLinks.ts [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+  const { isWatching } = colonySubscription;
+  const { colonyAddress, metadata } = colony || {};
+
+  const externalLinks = metadata?.externalLinks
+    ? sortExternalLinks(metadata.externalLinks)
+    : [];
+
+  const menuItems: DropdownMenuGroup[] = [
+    {
+      key: '1',
```
Comment (rdig) [inline src/components/v5/frame/ColonyHome/partials/DashboardHeader/partials/ColonyLinks/useHeaderLinks.ts:45]: You mentioned there was a PR where we would tackle all of these "keys", right ?

**Review by rdig — APPROVED** [review]
Left mostly nitpick comments or notes. But all in all stellar work!

Screenshot from 2024-01-16 19-26-42

File: src/routes/Routes.tsx [hunk 2]
```diff
@@ -139,7 +139,7 @@ const Routes = () => {
 
         {/* Colony routes */}
         <Route path={COLONY_HOME_ROUTE} element={<ColonyRoute />}>
-          <Route index element={<ColonyHome />} />
+          <Route index element={<ColonyHomePage />} />
```
[the remark this answers is not in the document]
Comment (jakubcolony) [reply src/routes/Routes.tsx:142]: Old routes should go away soon: #1708

**Review by jakubcolony — APPROVED** [review]
Very nice, reads much better

File: src/components/v5/frame/ColonyHome/partials/DashboardHeader/partials/ColonyLinks/useHeaderLinks.ts (hunk 1 above)
Comment (bassgeta) [reply src/components/v5/frame/ColonyHome/partials/DashboardHeader/partials/ColonyLinks/useHeaderLinks.ts:45]: nice catch, will do them now after I've also rebased this PR on master!

File: src/components/v5/frame/ColonyHome/partials/DashboardHeader/DashboardHeader.tsx [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+import useColonyContext from '~hooks/useColonyContext';
+import { formatText } from '~utils/intl';
+import { multiLineTextEllipsis } from '~utils/strings';
+import NativeTokenPill from '~v5/common/NativeTokenPill';
+
+import ColonyLinks from './partials/ColonyLinks';
+
+const displayName = 'v5.frame.ColonyHome.DashboardHeader';
+
+const MAX_DESCRIPTION_LENGTH = 250;
```
[the remark this answers is not in the document]
Comment (bassgeta) [reply src/components/v5/frame/ColonyHome/partials/DashboardHeader/DashboardHeader.tsx:12]: yeah I think a prop with default isn't bad at all, I can tackle that

**Review by jakubcolony — DISMISSED** [review]

File: src/components/v5/frame/ColonyHome/partials/DashboardHeader/partials/ColonyLinks/useHeaderLinks.ts [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+    isCopied: isShareUrlItemCopied,
+  } = useCopyToClipboard(5000);
+  const {
+    handleClipboardCopy: handleColonyAddressItemCopy,
+    isCopied: isColonyAddressItemCopied,
+  } = useCopyToClipboard(5000);
+
+  const colonyUrl = `${window.location.host}${pathname}`;
+  const { isWatching } = colonySubscription;
+  const { colonyAddress, metadata } = colony || {};
```
Comment (ArmandoGraterol) [inline src/components/v5/frame/ColonyHome/partials/DashboardHeader/partials/ColonyLinks/useHeaderLinks.ts:37]: With the changes to colony context this `|| {}` should no longer be required 🎉

**Review by ArmandoGraterol — APPROVED** [review]
Still looking good to me 👍

**Review by rdig — APPROVED** [review]

**Pull request merged**